src/BudgetManager.Shared/BudgetManager.Shared.DataAccess.MongoDB/BaseImplementation/BaseRepository.cs

```diff
@@ -19,9 +19,9 @@ public BaseRepository(IMongoDbSettings settings, IMongoClient client)
             _collection = database.GetCollection<TDocument>(GetCollectionName(typeof(TDocument)));
         }
 
-        public virtual Task<List<TDocument>> GetAllAsync(CancellationToken cancellationToken)
+        public virtual async Task<IEnumerable<TDocument>> GetAllAsync(CancellationToken cancellationToken)
         {
-            return Task.Run(() => _collection.Find(_ => true).ToListAsync(cancellationToken));
+            return await _collection.Find(_ => true).ToListAsync(cancellationToken);
```

Comment:
we need to make every Find -> FindAsync

Reply:
Can we make it?

Reply:
No, it is not possible, ToListAsync is only way to make it asynchronous